wasm-rpc-stubgen/src/lib.rs

```diff
@@ -435,3 +420,109 @@ pub fn initialize_workspace(
+            );
+            Ok(false)
+        }
+    }
+
+    pub(crate) fn find_world_name(unresolved_package: UnresolvedPackage) -> anyhow::Result<String> {
+        // In reality, there is only 1 interface in generated stub in 1 _stub.wit
+        for (_, interface) in unresolved_package.interfaces {
+            if let Some(name) = interface.name {
+                if name.starts_with("stub-") {
+                    let world_name = name.replace("stub-", "");
+                    return Ok(world_name);
+                }
+            }
+        }
+
+        Err(anyhow!("Failed to find world name from the stub. The interface name in stub is expected to have the pattern stub-<world-name>"))
+    }
+
+    pub(crate) fn dest_owns_stub_world(
+        stub_world_name: &str,
+        destination_wit_root: &UnresolvedPackage,
+    ) -> bool {
+        destination_wit_root
+            .worlds
+            .iter()
+            .map(|(_, world)| world.name.clone())
+            .collect::<Vec<_>>()
+            .contains(&stub_world_name.to_string())
+    }
+
+    // When copying the wit files of the target to the packages wit/deps in the source, we need to ensure
+    // these dependencies are not the source itself, or it's stub version
+    // For cases where adding the stub dependency to its own package is valid (i.e, in case of self-loop/ direct-cycle dependency)
+    // this function is/should-never-be called because, in this case destination owns the stub world (the stub to be copied) already
+    // and forms a different branch of logic.
+    pub(crate) fn is_invalid_dependency(
+        destination_wit_root: &UnresolvedPackage,
+        dependency_package: &UnresolvedPackage,
+    ) -> bool {
+        let self_stub_name = format!("{}-stub", destination_wit_root.name.name);
+
+        dependency_package.name == destination_wit_root.name
+            || (dependency_package.name.namespace == destination_wit_root.name.namespace
+                && dependency_package.name.name == self_stub_name)
+    }
+
+    // For those dependencies we add to the source, if they are importing from the skipped/invalid dependencies
+    // we simply make sure to delete them
+    pub(crate) fn replace_self_imports_from_dependencies(
+        dependency_wit_path: &PathBuf,
+        destination_wit_root: &UnresolvedPackage,
+    ) -> anyhow::Result<String> {
+        let read_data = fs::read_to_string(dependency_wit_path)?;
+        let self_stub_package = format!(
+            "{}:{}-stub",
+            destination_wit_root.name.namespace, destination_wit_root.name.name
+        );
+
+        let re = Regex::new(
+            format!(
+                r"import\s+{}(/[^;]*)?;",
+                regex::escape(self_stub_package.as_str())
+            )
+            .as_str(),
+        )
+        .unwrap();
+
+        Ok(re.replace_all(&read_data, "").to_string())
+    }
+
+    pub fn get_file_name(path: &PathBuf) -> anyhow::Result<String> {
```

Comment:
Not blocking at all, but there's no need to use String here, we can just use the PathBuf. I think we'd need to update the enum case `WitAction::WriteWit` to use PathBuf as a field instead of String.

Reply:
and take `&Path` as input